templates/workspace_services/azureml/terraform/main.tf

```diff
@@ -1,29 +1,61 @@
-resource "azurerm_machine_learning_workspace" "aml_workspace" {
-  name                          = local.workspace_name
-  resource_group_name           = data.azurerm_resource_group.ws.name
-  location                      = data.azurerm_resource_group.ws.location
-  application_insights_id       = azurerm_application_insights.ai.id
-  container_registry_id         = azurerm_container_registry.acr.id
-  friendly_name                 = var.display_name
-  description                   = var.description
-  high_business_impact          = true
-  key_vault_id                  = data.azurerm_key_vault.ws.id
-  public_network_access_enabled = var.is_exposed_externally ? true : false
-  storage_account_id            = azurerm_storage_account.aml.id
-  tags                          = local.tre_workspace_service_tags
-
-  identity {
-    type = "SystemAssigned"
+resource "azapi_resource" "aml_workspace" {
+  type      = "Microsoft.MachineLearningServices/workspaces@2024-10-01-preview"
+  name      = local.workspace_name
+  location  = data.azurerm_resource_group.ws.location
+  parent_id = data.azurerm_resource_group.ws.id
+  tags      = local.tre_workspace_service_tags
+
+  lifecycle { ignore_changes = [tags] }
+
+
+  dynamic "identity" {
+    for_each = var.enable_cmk_encryption ? [] : [1]
+    content {
+      type = "SystemAssigned"
+    }
+  }
+
+
+  dynamic "identity" {
+    for_each = var.enable_cmk_encryption ? [1] : []
+    content {
+      type         = "SystemAssigned, UserAssigned"
+      identity_ids = [data.azurerm_user_assigned_identity.ws_encryption_identity[0].id]
+    }
   }
 
-  lifecycle {
```

Comment:
Why are you removing the lifecycle part?

Reply:
It's on line 8, but, yes, should add `imageBuildCompute`, not sure why public_network_access is on there, Will test it out. Thanks.